multidim_threshold/learn.py

```diff
@@ -55,7 +56,8 @@ def weightedbinsearch(r: Rec, robust, eps=0.01) -> (array, array, array):
         fmid = flo if frhi < 0 else fhi
         return flo, fmid, fhi
 
-    while hi - lo > eps:
+
+    while (np.array(f(hi)) - np.array(f(lo)) > np.array(eps)).any():
```

Comment:
Same comments as binary search.
In fact, I feel the the two search algorithms can be combined, but maybe that'll over complicate things.